packages/maskbook/src/plugins/Valuables/api.tsx

```diff
@@ -0,0 +1,39 @@
+import { VCENT_BASE_URL } from './constants'
+
+export async function getTweetBid(id: string) {
+    // doublecheck regex testing
+    if (!/\d+/.test(id)) return
+
+    const response = await fetch(`${VCENT_BASE_URL}${id}`)
+    const { results } = (await response.json()) as {
+        results: tweetBids[]
+    }
+
+    if (results.length === 0) return
+
+    // return results[0].amount_usd
+    return {
+        amount_eth: results[0].amount_eth,
+        amount_usd: results[0].amount_usd,
+    }
+}
+
+/*
+  v.cent API's CORS policy isn't entirely clear to me yet. Looks to me like requests from twitter.com (or maybe my requests are from localhost) are getting no Access-Control-Allow-Origin header from the server.
```

Comment:
Maybe we need to add a proxy server for this. cc @yisiliu or @septs?

Reply:
Yeah I thought about that, looked at this one for a template: https://github.com/Rob--W/cors-anywhere.

Reply:
plz use proxy service, replace it.

```
https://vcent-agent.r2d2.to/data/tweet-txn?tweetID=
```